src/StreamJsonRpc/MessagePackFormatter.cs

```diff
@@ -1324,22 +1407,34 @@ public JsonRpcMessage Deserialize(ref MessagePackReader reader, MessagePackSeria
                 int propertyCount = readAhead.ReadMapHeader();
                 for (int i = 0; i < propertyCount; i++)
                 {
-                    string propertyName = readAhead.ReadString();
-                    if (propertyName == MethodPropertyName)
-                    {
-                        return options.Resolver.GetFormatterWithVerify<Protocol.JsonRpcRequest>().Deserialize(ref reader, options);
-                    }
-                    else if (propertyName == ResultPropertyName)
+                    // We read the property name in this fancy way in order to avoid paying to decode and allocate a string when we already know what we're looking for.
+                    ReadOnlySpan<byte> stringKey = MessagePack.Internal.CodeGenHelpers.ReadStringSpan(ref readAhead);
+                    switch (stringKey.Length)
                     {
-                        return options.Resolver.GetFormatterWithVerify<Protocol.JsonRpcResult>().Deserialize(ref reader, options);
-                    }
-                    else if (propertyName == ErrorPropertyName)
-                    {
-                        return options.Resolver.GetFormatterWithVerify<Protocol.JsonRpcError>().Deserialize(ref reader, options);
-                    }
+                        case 6:
```

Comment:
Since we are not using length to get the ulong from GetKey method do we need to separate out cases by length? Is there any harm in calling GetKey if the length is >6?

Reply:
`GetKey` supports any non-zero length, but it can only return a fully representative `ulong` for lengths of 8 or less.
We check length first because there's no reason to call `GetKey` at all if the length doesn't match one that we expect.